_posts/2015-12-13-weak-tests-hurt.markdown

```diff
@@ -0,0 +1,85 @@
+---
+layout: post
+title:  "Weak Tests Hurt"
+date:   2015-11-15 10:00:00
+categories: git development
+author: kent.safranski
+cover: /assets/images/covers/failed-test.jpg
+description: Writing tests isn't exactly an art but there are some really common mistakes made when writing them that can cost you big in the end.
+---
+
+Hopefully you're writing tests for your code, if not you really should. I don't look at testing as an exact science; it's something that needs to be done, but sometimes the happy-path (i.e. having a pasing suite of tests) becomes the goal instead of focusing on what the tests _actually test_.
+
+> I should be able to _ever so slightly_ alter a test and make it fail.
+
+If the above statement isn't true you should probably reevaluate your tests. Tests should ultimately be brittle. They, unlike your code, should perform one, single task and exit; pass/fail.
+
+A good test is one that can fail easier than it can pass.
+
+### An Example
+
+Let's say you're retrieving data from a database. Pretty standard CRUD stuff. For our example we'll retrieve a list of users:
+
+{% highlight javascript %}
```

Comment:
If we switch markdown processors to Kramdown we can get fenced code blocks, a al https://github.com/jhabdas/habd.as/blob/master/_posts/2015-04-02-tame-async-javascript-es6.md. But rather than doing that now I'd recommend waiting until we add deep link support (as it'll also require us to use Kramdown to properly output heading tags with an `id` attribute).